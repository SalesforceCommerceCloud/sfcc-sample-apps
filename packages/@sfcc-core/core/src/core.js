/*
    Copyright (c) 2020, salesforce.com, inc.
    All rights reserved.
    SPDX-License-Identifier: BSD-3-Clause
    For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
*/
//
// SFRA Core Code
//

export const LOGGER_KEY = Symbol('Logger Service');
export const API_EXTENSIONS_KEY = Symbol('API Extensions');

class Core {

    // singletons (logger, etc)
    get services() {
        return this._services;
    }

    // multi-values
    get extensions() {
        return this._factoryExtensions;
    }

    /**
     * Services can have only a single entry/instance per service name.
     * @param name
     * @param service
     */
    registerService(key, service) {
        this.logger.log(`registerService(${key.toString()})`);

        if (this._factoryServices[key]) {
            throw new Error(`Service,'${key}', already registered`);
        }
        this._factoryServices[key] = service;
    }

    /**
     * Extensions can have multiple entries per extension key.
     * e.g. extensions['payment'] = [fakePayment, paypal, applePay];
     * @param key
     * @param extension
     */
    registerExtension(key, extension) {
        this.logger.log(`registerExtension(${key.toString()})`);

        if (!this._factoryExtensions[key]) {
            this._factoryExtensions[key] = [];
        }
        this._factoryExtensions[key].push(extension);
    }

    /**
     * Creates instances from registered extensions factories
     *
     * @param key The specific extension(s) to instantiate. If undefined Otherwise instantiate all.
     */
    initializeExtensions(key) {
        this.logger.log(`initializeExtensions(${key.toString()})`);

        const keys = (key) ? [key] : Object.getOwnPropertySymbols(this._factoryExtensions);

        keys.forEach(key => {
            if (key && !!this._extensions[key]) {
                const msg = `Error: ${key.toString()} extensions already initialized`;
                core.logger.error(msg);
                throw new Error(msg);
            }
            this._extensions[key] = [];
            this.getExtension(key).forEach(extension => {
                // instantiate extension
                this._extensions[key].push(extension());
            })
        })
    }

    /**
     * Getting service[s] will lazily instantiate the service.
     * @param servicekey
     * @return {*}
     */
    getService(key) {
        if (this._services[key]) {
            return this._services[key];
        } else if (this._factoryServices[key]) {
            // create the service instance
            return this._services[key] = new this._factoryServices[key]();
        } else if ( key === LOGGER_KEY ){
            // A logger isn't registered yet
            return console;
        } else {
            throw new Error(`Service ${key.toString()} does not exist`);

        }
    }

    get logger() {
        return this.getService(LOGGER_KEY);
    }

    /**
     * Getting extension[s] will return extensions' factories.
     * (Yes plural and possessive)
     * @param extensionKey
     * @return {*}
     */
    getExtension(key) {
        return this._factoryExtensions[key];
    }

    constructor() {
        // just to debug
        this.INSTANCE = 'Core Instance: ' + new Date().getTime();

        // Instance maps for services and extensions
        this._services = {};
        this._extensions = {}; // unused at the moment

        // Factory maps for services and extensions
        this._factoryServices = {};
        this._factoryExtensions = {};
        this.logger.log('Create the Core instance used to register services and extensions.');
    }
}

const coreSingleton = new Core();
export const core = coreSingleton;
