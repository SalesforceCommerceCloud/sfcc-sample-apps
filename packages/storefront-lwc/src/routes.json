[
  {
    "id": "home",
    "path": "/",
    "isRoot": true,
    "view": "home",
    "label": "Home",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "home"
      }
    }
  },
  {
    "id": "search",
    "path": "/search/:query",
    "isRoot": false,
    "view": "search",
    "label": "Search",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "search"
      }
    }
  },
  {
    "id": "product",
    "path": "/product/:pid",
    "isRoot": false,
    "view": "product",
    "label": "Product",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "product"
      }
    }
  },
  {
    "id": "register",
    "path": "/register",
    "isRoot": false,
    "view": "login",
    "label": "LoginRegister",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "register"
      }
    }
  },
  {
    "id": "login",
    "path": "/login",
    "isRoot": false,
    "view": "login",
    "label": "Login",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "login"
      }
    }
  },
  {
    "id": "cart",
    "path": "/cart",
    "isRoot": false,
    "view": "cart",
    "label": "Cart",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "cart"
      }
    }
  },
  {
    "id": "about",
    "path": "/about",
    "view": "about",
    "label": "About",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "about"
      }
    }
  },
  {
    "id": "error",
    "path": "/error",
    "isDefault": true,
    "view": "error",
    "label": "Error",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "error"
      }
    }
  },
  {
    "id": "styleguide",
    "path": "/styleguide",
    "isRoot": false,
    "view": "styleguide",
    "label": "styleguide",
    "page": {
      "type": "comm_namedPage",
      "attributes": {
        "pageName": "styleguide"
      }
    }
  }
]
