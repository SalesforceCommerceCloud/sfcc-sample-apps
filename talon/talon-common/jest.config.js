module.exports = {
    displayName: "talon-common",
    moduleFileExtensions: [
      "js", "json"
    ],
    moduleDirectories: [
      "node_modules",
      "src"
    ],
    testEnvironment: "node",
    testURL: "https://localhost",
    testRegex: "/__tests__/.*.test.js$"
};