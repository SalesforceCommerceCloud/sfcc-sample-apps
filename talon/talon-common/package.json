{
  "_args": [
    [
      "talon-common@0.0.1",
      "/Volumes/Share/sf/talon-commerce"
    ]
  ],
  "_from": "talon-common@0.0.1",
  "_id": "talon-common@0.0.1",
  "_inBundle": false,
  "_integrity": "sha512-EJhH9MdST254Kq4NeqgvN+FbOgWOwwgESBCXIJZV3PTJ2kP/Sv1bBS4EjPACmSK0INafoRcGeayXItKijFnWgQ==",
  "_location": "/talon-common",
  "_phantomChildren": {},
  "_requested": {
    "type": "version",
    "registry": true,
    "raw": "talon-common@0.0.1",
    "name": "talon-common",
    "escapedName": "talon-common",
    "rawSpec": "0.0.1",
    "saveSpec": null,
    "fetchSpec": "0.0.1"
  },
  "_requiredBy": [
    "/"
  ],
  "_resolved": "https://nexus.soma.salesforce.com/nexus/content/groups/npm-all/talon-common/-/talon-common-0.0.1.tgz",
  "_spec": "0.0.1",
  "_where": "/Volumes/Share/sf/talon-commerce",
  "dependencies": {
    "@babel/plugin-transform-regenerator": "^7.0.0",
    "@babel/preset-env": "^7.3.1"
  },
  "devDependencies": {
    "@babel/core": "^7.2.2",
    "@babel/plugin-transform-modules-commonjs": "^7.2.0",
    "babel-jest": "^24.1.0",
    "jest": "^24.1.0"
  },
  "gitHead": "330cb3e3a6655c67dd69790be0432aea291460e8",
  "main": "dist/es.js",
  "module": "dist/es.js",
  "name": "talon-common",
  "scripts": {
    "build": "cross-env rollup -c ./scripts/rollup.config.js",
    "clean": "rm -rf dist",
    "test": "jest",
    "test:debug": "node --inspect-brk ./node_modules/.bin/jest --runInBand"
  },
  "version": "0.0.1"
}
