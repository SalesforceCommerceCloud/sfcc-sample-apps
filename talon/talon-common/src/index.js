export { getOutputConfigs } from './output-configs';

export {
    getResourceUrl,
    parseUrl,
    resourceDescriptorToString,
    parseResourceDescriptor,
    RESOURCE_TYPES
} from './utils/resource-utils';

export { moduleSpecifierToElementName, elementNameToModuleSpecifier, moduleSpecifierToId, convertToKebabCase } from './utils/naming-utils';

export { getViewModuleName, getViewModuleFullyQualifiedName, VIEW_NAMESPACE } from './utils/view-utils';

export { compatBabelOptions } from './compat-babel-options';