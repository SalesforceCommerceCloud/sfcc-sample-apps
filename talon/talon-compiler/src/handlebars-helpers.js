/**
 * The JavaScript code that we generate here is sent to the client as is
 * and does not go through babel. For this reason it need to work in IE 11
 * and older browser.
 *
 * To enforce that we have some tests in handlebars-helpers.test.js that will
 * extract the generated JavaScript, feed it to babel and make sure the output matches.
 *
 * If you add or modify generated JavaScript here make sure you run handlebars-helpers.test.js
 * and update the generated code if needed.
 */

const Handlebars = require('handlebars');
const { assert } = require('./utils/assert');
const { getResourceUrl } = require('talon-common');
const { getBrandingPropertyValue } = require('./metadata/process-branding');

/**
 * Generates the script tags necessary to load the Talon framework,
 * the components components needed to load the root route,
 * and configure the framework e.g. set base path, register the
 * routes and set branding properties.
 *
 * @param {Object} context - The context
 * @param {Object} context.brandingProperties - The branding properties
 * @param {Object} context.routes - The routes definition
 * @param {string} context.currentRoute - The current route for which the HTML document is generated
 * @param {string} context.mode - The compile mode i.e. 'dev', 'prod', 'compat' or 'prod_compat'
 * @param {string} context.basePath - Base path to prepend to generated URLs
 * @param {object} context.resources - Mapping between resource descriptors and UIDs
 * @param {object} context.theme - The theme
 * @param {object} context.viewToThemeLayoutMap - Mapping between views and themeLayouts
 * @param {string} context.sourceNonce sourceNonce - The UUID for script-src CSP nonce
 * @param {object} context.user - The context user
 */
function talonInit({ brandingProperties, routes, currentRoute, mode, basePath, locale, resources, theme, viewToThemeLayoutMap, sourceNonce, isPreview, user }) {
    function getUrl(descriptor) {
        return getResourceUrl(descriptor, mode, resources[descriptor]);
    }

    assert(theme, "No theme specified");
    assert(currentRoute, "Current route not specified");
    assert(mode, "Mode not specified");

    const frameworkUrl = getUrl('framework://talon');
    const designtimeUrl = getUrl('framework://talondesign');

    const designtimeUrlOut = isPreview ? `<script src="${basePath}${designtimeUrl}"></script>\n` : '';

    const out = `<script src="${basePath}${frameworkUrl}"></script>\n${designtimeUrlOut}
    <script type="text/javascript" ${sourceNonce ? `nonce="${sourceNonce}"` : ""}>
    Talon.configProvider.register({
        getBasePath: function getBasePath() {
            return ${JSON.stringify(basePath)};
        },
        getMode: function getMode() {
            return ${JSON.stringify(mode)};
        },
        getLocale: function getLocale() {
            return ${JSON.stringify(locale)};
        },
        getUser: function getUser() {
            return ${JSON.stringify(user)};
        }
    });
    Talon.themeService.setTheme(${JSON.stringify(theme)});
    Talon.themeService.setViewToThemeLayoutMap(${JSON.stringify(viewToThemeLayoutMap)});
    Talon.routingService.registerRoutes(${JSON.stringify(routes)});
    ${talonBranding(brandingProperties)}
    ${Object.keys(resources).length > 0 ? `Talon.moduleRegistry.setResourceUids(${JSON.stringify(resources)});` : ""}
</script>`;

    return new Handlebars.SafeString(out);
}

Handlebars.registerHelper('talonInit', talonInit);

/**
 * Generates the script tags necessary to load the Talon framework,
 * the components components needed to load the root route,
 * and configure the framework e.g. set base path, register the
 * routes and set branding properties.
 *
 * @param {Object} context - The context
 * @param {Object} context.brandingProperties - The branding properties
 * @param {Object} context.routes - The routes definition
 * @param {string} context.currentRoute - The current route for which the HTML document is generated
 * @param {string} context.mode - The compile mode i.e. 'dev', 'prod', 'compat' or 'prod_compat'
 * @param {string} context.basePath - Base path to prepend to generated URLs
 * @param {object} context.resources - Mapping between resource descriptors and UIDs
 * @param {object} context.theme - The theme
 * @param {object} context.viewToThemeLayoutMap - Mapping between views and themeLayouts
 */
function talonViews({ currentRoute, mode, basePath, locale, resources, viewToThemeLayoutMap }) {
    function getUrl(descriptor) {
        return getResourceUrl(descriptor, mode, resources[descriptor]);
    }

    // load theme layout and component for the current route
    const themeLayout = viewToThemeLayoutMap[currentRoute.view];
    const themeLayoutUrl = getUrl(`view://${themeLayout}@${locale}`);
    const currentViewUrl = getUrl(`view://${currentRoute.view}@${locale}`);
    const out = `<script src="${basePath}${themeLayoutUrl}"></script>\n<script src="${basePath}${currentViewUrl}"></script>\n`;

    return new Handlebars.SafeString(out);
}

Handlebars.registerHelper('talonViews', talonViews);

/**
 * Generates the script that will create the theme layout component and add it to the DOM.
 *
 * @param {string} context.sourceNonce sourceNonce - The UUID for script-src CSP nonce
 * @param {string} containerId - ID of the DOM container in which the theme layout component will be rendered
 */
function talonApp({sourceNonce}, containerId) {
    const appContainer = "talon/app";

    const out = `<script type="text/javascript" ${sourceNonce ? `nonce="${sourceNonce}"` : ""}>
    Talon.componentService.createElement("${appContainer}").then(function(element) {
        document.getElementById("${containerId}").appendChild(element);
    });
</script>`;

    return new Handlebars.SafeString(out);
}

Handlebars.registerHelper('talonApp', talonApp);

/**
 * Generates the styles for the given branding properties.
 *
 * @param {Object} brandingProperties - The branding properties
 */
function talonBranding(brandingProperties) {
    const propsOut = {};
    for (let i = 0; i < brandingProperties.length; i++) {
        const prop = brandingProperties[i];
        propsOut[prop.name] = getBrandingPropertyValue(prop);
    }

    return `Talon.brandingService.setBrandingProperties(${JSON.stringify(propsOut)});`;
}