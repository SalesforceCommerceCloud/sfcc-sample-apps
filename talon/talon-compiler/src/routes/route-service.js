const { resourceDescriptorToString, RESOURCE_TYPES } = require('talon-common/dist/cjs');
const { getView, getAllViews } = require('../metadata/metadata-service');
const TALON_RESOURCES = require('talon-framework/dist/resources.json');
const TALON_RESOURCE_DESCRIPTORS = [...Object.keys(TALON_RESOURCES)];
const pathToRegexp = require('path-to-regexp');

// TODO remove
const IGNORED_RESOURCES = [
    "view://customPerf"
];

const DESIGNTIME_RESOURCE = "framework://talondesign";

/**
 * Get the resources the given routes depend on.
 *
 * @param  {Object|Object[]} routes the route or routes to get the resources for
 * @returns an array containing the resource descriptors
 *          the given routes depend on
 */
function getRoutesResources(routes, theme, locale, isPreview = false) {
    const routeViewDescriptors = getAllViews(routes, theme).map(view => {
        return resourceDescriptorToString({ type: RESOURCE_TYPES.VIEW, name: view.name, locale });
    });

    //
    const resources = [
        ...TALON_RESOURCE_DESCRIPTORS,
        ...routeViewDescriptors
    ];

    // remove duplicates and filter resources
    const result = [...new Set(resources)].filter(resourceDescriptor => {
        const isDesigntimeResourceIgnored = !isPreview && DESIGNTIME_RESOURCE === resourceDescriptor;
        const isResourceIgnored = IGNORED_RESOURCES.includes(resourceDescriptor.split('@')[0]);
        return !isDesigntimeResourceIgnored && !isResourceIgnored;
    });

    return result;
}

/**
 * Get the route matching the given path.
 *
 * @param {Object[]} routes The routes to get the current route from
 * @param {string} path The path, without basePath
 * @returns {Object} The route matching the given path, or the default route
 */
function getRouteMatchingPath(routes, path) {
    const currentRoute = path && routes.find(route => {
        const regexp = pathToRegexp(route.path, []);
        return regexp.test(path);
    });
    return currentRoute || routes.find(route => route.isDefault);
}

/**
 * Get all views which are routed
 */
function getViewToThemeLayoutMap(routes, theme) {
    const viewToThemeLayoutMap = {};
    routes.forEach((route) => {
        const view = getView(route.view);
        const themeLayout = theme.themeLayouts[view.themeLayoutType];
        viewToThemeLayoutMap[route.view] = themeLayout.view;
    });
    return viewToThemeLayoutMap;
}

module.exports = { getRoutesResources, getRouteMatchingPath, getViewToThemeLayoutMap };