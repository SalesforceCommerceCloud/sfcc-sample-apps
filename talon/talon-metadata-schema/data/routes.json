[
    {
        "name": "home",
        "path": "/",
        "isRoot": true,
        "view": "home"
    },
    {
        "name": "confirmation",
        "path": "/confirmation",
        "view": "confirmation"
    },
    {
        "name": "about",
        "path": "/about",
        "view": "about"
    },
    {
        "name": "community_flashhelp_customlwc_perf",
        "path": "/community_flashhelp_customlwc_perf",
        "view": "customPerf"
    },
    {
        "name": "community_user_login",
        "path": "/login",
        "view": "login"
    },
    {
        "name": "faq",
        "path": "/faq",
        "isDefault": false,
        "view": "faq"
    },
    {
        "name": "error",
        "path": "/error",
        "isDefault": true,
        "view": "error"
    }
]