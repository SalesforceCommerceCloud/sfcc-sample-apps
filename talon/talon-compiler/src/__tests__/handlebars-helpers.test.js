import '../handlebars-helpers';
import { compatBabelOptions } from 'talon-common';
import { getViewToThemeLayoutMap } from '../routes/route-service';
import { helpers } from 'handlebars';
import { transform } from '@babel/core';
import beautify from 'js-beautify';

const routes = require('./fixtures/src/routes.json');
const currentRoute = routes[0];
const brandingProperties = require('./fixtures/src/branding.json');
const theme = require('./fixtures/src/theme.json');
const viewToThemeLayoutMap = getViewToThemeLayoutMap(routes, theme);

jest.mock('../metadata/metadata-service', () => ({
    getView: () => {
        return {
            "themeLayoutType": "inner"
        };
    }
}));

/**
 * Extract inline scripts from HTML.
 */
function extractScripts(code) {
    const scripts = [];
    const regex = /<script[^>]*>([^<]+)<\/script>/g;
    let match;
    while ((match = regex.exec(code))) {
        scripts.push(match[1]);
    }
    return scripts;
}

describe('handlebars-helpers', () => {
    describe('talonInit', () => {
        const basePath = '/test';
        const mode = 'prod';
        const locale = 'fr';
        const resources = {};
        const sourceNonce = "T3stVa1ue";
        const user = { isGuest: false };

        it('returns script tags for talon init', () => {
            const { string: out } = helpers.talonInit({ theme, brandingProperties, routes, currentRoute, mode, basePath, resources, locale, viewToThemeLayoutMap, user });
            expect(out).toMatchSnapshot();
        });

        it('returns script tags for talon init with script-src nonce', () => {
            const { string: out } = helpers.talonInit({ theme, brandingProperties, routes, currentRoute, mode, basePath, resources, locale, viewToThemeLayoutMap, sourceNonce, user });
            out.includes('<script type="text/javascript" nonce="T3stVa1ue">');
            expect(out).toMatchSnapshot();
        });

        it('throws when no theme specified', () => {
            expect(() => {
                helpers.talonInit({ brandingProperties, routes, currentRoute, mode, basePath, resources, locale });
            }).toThrow();
        });

        it('throws when no mode specified', () => {
            expect(() => {
                helpers.talonInit({ theme, brandingProperties, routes, currentRoute, basePath, resources, locale });
            }).toThrow();
        });

        [undefined, null, ""].forEach((invalidTestRoute) => {
            it(`throws when current route is ${JSON.stringify(invalidTestRoute)}`, () => {
                expect(() => {
                    helpers.talonInit({ theme, brandingProperties, routes, currentRoute: invalidTestRoute, mode, basePath, resources, locale });
                }).toThrow('Current route not specified');
            });
        });

        it('returns es5 javascript', () => {
            const { string: out } = helpers.talonInit({ theme, brandingProperties, routes, currentRoute, mode, basePath, resources, locale, viewToThemeLayoutMap, sourceNonce, user });
            const scripts = extractScripts(out);

            expect(scripts).toHaveLength(1);

            const { code: es5Script } = transform(scripts[0], compatBabelOptions);
            expect(beautify(scripts[0])).toBe(beautify(es5Script));
        });

        it('returns designtime script for preview', () => {
            const { string: out } = helpers.talonInit({ theme, brandingProperties, routes, currentRoute, mode, basePath, resources, locale, viewToThemeLayoutMap, sourceNonce, isPreview: true });
            expect(out).toMatchSnapshot();
        });
    });

    describe('talonViews', () => {
        const basePath = '/test';
        const mode = 'prod';
        const locale = 'fr';
        const resources = {};
        const sourceNonce = "T3stVa1ue";

        it('returns script tags for talon views', () => {
            const { string: out } = helpers.talonViews({ theme, brandingProperties, routes, currentRoute, mode, basePath, resources, locale, viewToThemeLayoutMap });
            expect(out).toMatchSnapshot();
        });

        it('returns es5 javascript', () => {
            const { string: out } = helpers.talonViews({ theme, brandingProperties, routes, currentRoute, mode, basePath, resources, locale, viewToThemeLayoutMap, sourceNonce });
            const scripts = extractScripts(out);
            expect(scripts).toHaveLength(0);
        });
    });

    describe('talonApp', () => {
        const sourceNonce = "T3stVa1ue";
        const testContainerId = "01234";
        const testContext = { sourceNonce };

        it('returns script tags for talon app component', () => {
            const { string: out } = helpers.talonApp({}, testContainerId);
            out.includes('document.getElementById("01234"');
            expect(out).toMatchSnapshot();
        });

        it('returns script tags for talon app component with script-src nonce', () => {
            const { string: out } = helpers.talonApp(testContext, testContainerId);
            out.includes('<script type="text/javascript" nonce="T3stVa1ue">');
            expect(out).toMatchSnapshot();
        });

        it('returns es5 javascript', () => {
            const { string: out } = helpers.talonApp({ sourceNonce }, testContainerId);
            const scripts = extractScripts(out);

            expect(scripts).toHaveLength(1);

            const { code: es5Script } = transform(scripts[0], compatBabelOptions);
            expect(beautify(scripts[0])).toBe(beautify(es5Script));
        });
    });
});