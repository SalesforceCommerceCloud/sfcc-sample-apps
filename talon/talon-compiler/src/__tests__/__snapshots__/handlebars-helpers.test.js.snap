// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handlebars-helpers talonApp returns script tags for talon app component 1`] = `
"<script type=\\"text/javascript\\" >
    Talon.componentService.createElement(\\"talon/app\\").then(function(element) {
        document.getElementById(\\"01234\\").appendChild(element);
    });
</script>"
`;

exports[`handlebars-helpers talonApp returns script tags for talon app component with script-src nonce 1`] = `
"<script type=\\"text/javascript\\" nonce=\\"T3stVa1ue\\">
    Talon.componentService.createElement(\\"talon/app\\").then(function(element) {
        document.getElementById(\\"01234\\").appendChild(element);
    });
</script>"
`;

exports[`handlebars-helpers talonInit returns designtime script for preview 1`] = `
"<script src=\\"/test/talon/framework/latest/prod/talon.js\\"></script>
<script src=\\"/test/talon/framework/latest/prod/talondesign.js\\"></script>

    <script type=\\"text/javascript\\" nonce=\\"T3stVa1ue\\">
    Talon.configProvider.register({
        getBasePath: function getBasePath() {
            return \\"/test\\";
        },
        getMode: function getMode() {
            return \\"prod\\";
        },
        getLocale: function getLocale() {
            return \\"fr\\";
        },
        getUser: function getUser() {
            return undefined;
        }
    });
    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
    Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutInner\\"});
    Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
    Talon.brandingService.setBrandingProperties({\\"SomeProperty\\":\\"some value\\",\\"SomeMisc\\":\\"misc data\\",\\"MySelfie\\":\\"url(/all/the/angles.jpg)\\"});
    
</script>"
`;

exports[`handlebars-helpers talonInit returns script tags for talon init 1`] = `
"<script src=\\"/test/talon/framework/latest/prod/talon.js\\"></script>

    <script type=\\"text/javascript\\" >
    Talon.configProvider.register({
        getBasePath: function getBasePath() {
            return \\"/test\\";
        },
        getMode: function getMode() {
            return \\"prod\\";
        },
        getLocale: function getLocale() {
            return \\"fr\\";
        },
        getUser: function getUser() {
            return {\\"isGuest\\":false};
        }
    });
    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
    Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutInner\\"});
    Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
    Talon.brandingService.setBrandingProperties({\\"SomeProperty\\":\\"some value\\",\\"SomeMisc\\":\\"misc data\\",\\"MySelfie\\":\\"url(/all/the/angles.jpg)\\"});
    
</script>"
`;

exports[`handlebars-helpers talonInit returns script tags for talon init with script-src nonce 1`] = `
"<script src=\\"/test/talon/framework/latest/prod/talon.js\\"></script>

    <script type=\\"text/javascript\\" nonce=\\"T3stVa1ue\\">
    Talon.configProvider.register({
        getBasePath: function getBasePath() {
            return \\"/test\\";
        },
        getMode: function getMode() {
            return \\"prod\\";
        },
        getLocale: function getLocale() {
            return \\"fr\\";
        },
        getUser: function getUser() {
            return {\\"isGuest\\":false};
        }
    });
    Talon.themeService.setTheme({\\"themeLayouts\\":{\\"inner\\":{\\"view\\":\\"themeLayoutInner\\"},\\"default\\":{\\"view\\":\\"themeLayoutDefault\\"}}});
    Talon.themeService.setViewToThemeLayoutMap({\\"home\\":\\"themeLayoutInner\\",\\"myPage\\":\\"themeLayoutInner\\",\\"error\\":\\"themeLayoutInner\\"});
    Talon.routingService.registerRoutes([{\\"name\\":\\"home\\",\\"path\\":\\"/\\",\\"isRoot\\":true,\\"view\\":\\"home\\"},{\\"name\\":\\"my-page\\",\\"path\\":\\"/my-page\\",\\"view\\":\\"myPage\\"},{\\"name\\":\\"error\\",\\"path\\":\\"/error\\",\\"isDefault\\":true,\\"view\\":\\"error\\"}]);
    Talon.brandingService.setBrandingProperties({\\"SomeProperty\\":\\"some value\\",\\"SomeMisc\\":\\"misc data\\",\\"MySelfie\\":\\"url(/all/the/angles.jpg)\\"});
    
</script>"
`;

exports[`handlebars-helpers talonViews returns script tags for talon views 1`] = `
"<script src=\\"/test/talon/view/latest/prod/fr/themeLayoutInner.js\\"></script>
<script src=\\"/test/talon/view/latest/prod/fr/home.js\\"></script>
"
`;
